site/src/api/api.ts

```diff
@@ -1,20 +1,36 @@
 /**
- * @fileoverview This file is imported externally by things like vscode-coder so
- * it must not import anything using aliases otherwise it will break when being
- * imported by those external consumers.  For example, `utils/delay` must be
- * imported using `../utils/delay` instead.
+ * @file Coder is starting to import the Coder API file into more and more
+ * external projects, as a "pseudo-SDK". We are not at a stage where we are
+ * ready to commit to maintaining a public SDK, but we need equivalent
+ * functionality in other places.
+ *
+ * Message somebody from Team Blueberry if you need more context, but so far,
+ * these projects are importing the file:
+ *
+ * - The Coder VS Code extension
+ *   @see {@link https://github.com/coder/vscode-coder}
+ * - The Coder Backstage plugin
+ *   @see {@link https://github.com/coder/backstage-plugins}
+ *
+ * It is important that this file not do any aliased imports, or else the other
+ * consumers could break (particularly for platforms that limit how much you can
+ * touch their configuration files, like Backstage). Relative imports are still
+ * safe, though.
+ *
+ * For example, `utils/delay` must be imported using `../utils/delay` instead.
  */
-
-import axios, { isAxiosError } from "axios";
+import globalAxios, { isAxiosError } from "axios";
 import type dayjs from "dayjs";
 import userAgentParser from "ua-parser-js";
 import { delay } from "../utils/delay";
 import * as TypesGen from "./typesGenerated";
 
+export const coderAxiosInstance = globalAxios.create();
```

Comment:
Should we call this `axios` instead? I feel like `coder` is implicit because of the repository, and `instance` could be implicit as well because you'd just use the global axios otherwise.

Reply:
So the main thing I was going for with the more specific name, was decreasing ambiguity/obscurity for what's going to end up being a relatively global value.

Before I made this PR, we weren't importing the global instance as `globalAxios` – we were just importing it as `axios`, and I feel like most people are accustomed to using the global instance, and not thinking about there being more than the singleton. So for somebody who starts working with the files months from now, is it going to be that obvious from a glance what the difference between these is? Not just that they're different, but why they're different?
```ts
import axios from "axios";
import { axios } from "./api";
```
Maybe I'm overthinking this, and I can change the name, but between the value being so global in the Coder codebase, and it now being used in multiple codebases, it feels like being a bit more pedantic with the name is worth it

Reply:
The ambiguity rationale is fair. Maybe we should settle somewhere in the middle. I'm good with `axiosInstance`.

My hesitancy to prefixing with `coder` is that there's always a scenario where that verbosity could seem reasonable, and I'd hate to introduce an inconsistency in our existing naming scheme. If a user is importing from our API, it's fair to assume it's scoped to our API, not globally.

Reply:
Yeah, I'm cool with splitting the difference